File: MAP/Laborator3/src/validator/ValidatorTema.java
package validator;

import domain.Tema;
import repository.ValidationException;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ValidatorTema implements Validator<Tema> {
    private static Pattern idPtrn = Pattern.compile("^[1-9][0-9]*$");
    private static Pattern datePtrn = Pattern.compile("^[1-9]|1[0-4]$");

    private void validateId(Integer id){
        String idS=id.toString();
        Matcher mtch = idPtrn.matcher(idS);
        if(!mtch.matches()){
            throw new ValidationException("Id incorect!");
        }
    }

    private void validateDate(Integer data){
        String dataS=data.toString();
        Matcher mtch = datePtrn.matcher(dataS);
        if(!mtch.matches()){
            throw new ValidationException("Data incorecta!");
        }
    }

    private void validateInterval(Integer predare,Integer deadline){
        if(predare>deadline)
            throw  new ValidationException("Date incorecte!");
    }


    @Override
    public void validate(Tema entity) throws ValidationException {
        validateId(entity.getID());
        validateDate(entity.getDeadline());
        validateDate(entity.getDataPredare());
        validateInterval(entity.getDataPredare(),entity.getDeadline());
    }
}

File: MAP/Laborator11-14/CatalogMAP/CatalogMAP/validator/ValidatorTema.cs
using CatalogMAP.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace CatalogMAP.validator
{
    public class ValidatorTema : IValidator<Tema>
    {
        private void ValidateId(string id)
        {
            Match mtch = Regex.Match(id, @"^[1-9][0-9]*$");
            if (!mtch.Success)
            {
                throw new ValidationException("Id incorect!");
            }
        }

        private void ValidateDate(string dataS)
        {
            Match mtch = Regex.Match(dataS, @"^[1-9]|1[0-4]$");
            if (!mtch.Success)
            {
                throw new ValidationException("Data incorecta!");
            }
        }

        
        private void ValidateInterval(string predare, string deadline)
        {
            if (Int32.Parse(predare) > Int32.Parse(deadline))
                throw new ValidationException("Date incorecte!");
        }

        public void Validate(Tema entity)
        {
            ValidateId(entity.ID);
            ValidateDate(entity.Deadline);
            ValidateDate(entity.DataPredare);
            ValidateInterval(entity.DataPredare, entity.Deadline);
        }
    }
}
